### ISSUE TITLE
Expected 1 space after colon with &:not(:first-child)

### ISSUE THREAD

**Issue by Sental** [magento/magento-coding-standard#395]
### Preconditions

1. In a less file if you use &:not(:first-child) it generates spacing warnings.

### Steps to reproduce

1.  Use &:not(:first-child) in a less file.
2. Run the coding standard from command line. `vendor/bin/phpcs --standard=Magento2 path/to/module`

### Expected result

1. No Warnings are generated.

### Actual result

1. Warnings are generated
154 | WARNING | Expected 1 space after colon in style definition; newline found
154 | WARNING | Expected 1 space after colon in style definition; 0 found

**Comment by mrtuvn** [comment]
Same issue experiences. Actually all pseudo classes not need require spacing after

**Comment by hostep** [comment]
The `phpcs` tools should only scan `*.php` (and maybe `*.phtml`) files, why is it being used to scan `*.less` files as well? That makes little sense?

**Comment by ihor-sviziev** [comment]
@sivaschenko, could you take a look?
We have an example where this issue reproducing on a PR - https://github.com/magento/magento2/pull/35101#issuecomment-1146916983

**Comment by sivaschenko** [comment]
@ihor-sviziev yes, we'll work on it

**Comment by ihor-sviziev** [comment]
I guess the changes are the following: https://github.com/magento/magento-coding-standard/commit/756ca2dff2e80013c13f6fde5049444cfd137871

**Comment by mrtuvn** [comment]
Also found magento yelling warning non-sense at this code

```less
.box-tocart {
    .paypal:first-of-type {
      margin-top: 13px;
    }
  }

```
:first-of-type totally valid. It's doesn't require space after that colon


```less
.lib-link(
                @_link-color: @primary__color,
                @_link-text-decoration: none,
                @_link-color-visited: @primary__color,
                @_link-text-decoration-visited: none,
                @_link-color-hover: @primary__color,
                @_link-text-decoration-hover: underline,
                @_link-color-active: @primary__color,
                @_link-text-decoration-active: underline
            );


```
Also detected new line after colon at mixins less declare => still valid with me in my point of view
